master/buildbot/worker/base.py

```diff
@@ -781,6 +781,8 @@ def buildFinished(self, sb):
         if not self.building:
             if self.build_wait_timeout == 0:
                 d = self.insubstantiate()
+                d.addCallback(lambda _:
+                              AbstractWorker.disconnect(self))
```

Comment:
This change makes sense to me to be added to insubstantiate, just not 100% sure if it should be kwarg flagged or not.

Also, I noticed insubstantiate() is calling `self.botmaster.maybeStartBuildsForWorker(self.name)` at the end, and we turn around and call that method again here. If we move the disconnect logic to insubstantiate can we drop the call to `self.botmaster.maybeStartBuildsForWorker(self.workername)` from this method?